core/src/org/javarosa/core/model/data/GeoPointData.java

```diff
@@ -132,4 +148,21 @@ public GeoPointData cast(UncastData data) throws IllegalArgumentException {
         }
         return new GeoPointData(ret);
     }
+
+    /**
+     * Truncate double to have the given number of decimals.
+     * @param x double to be truncated
+     * @param numberofDecimals number of decimals that should present in result
+     */
+    private static double truncateDecimal(double x, int numberofDecimals) {
+        // via:
+        // https://stackoverflow.com/questions/7747469/how-can-i-truncate-a-double-to-only-two-decimal-places-in-java/21468258#21468258
+        BigDecimal decimal;
```

Comment:
I don't think BigDecimal's are Java 1.3 compliant, btw, so we probably need to find a J2ME compatible way to do this truncation/rounding.